### DISCUSSION TITLE
[Q&A] 如何修改上传图片限制，以支持所有文件上传？

### DISCUSSION THREAD

**Discussion by cyx1322** [lobehub/lobehub#933] [Development Question] [answered]
### 🧐 问题描述 | Proposed Solution

我使用第三方的ChatGPT API，其支持通过gpt-4-vision(base64)的请求方式上传文件。
如何修改相关代码，调整上传策略，使得可以将图片之外的其他文件通过gpt-4-v上传呢？谢谢！

**Comment by arvinxx** [comment]
gpt-4v 就不支持文件，就算咱改成支持文件了，模型不支持又有啥用呢？

[a comment by an automated account is not shown]

**Comment by cyx1322** [comment]
我用的是第三方的API，是支持文件的，上传方法同图片，使用base64，所以想问一下从哪里能修改

[a comment by an automated account is not shown]

**Comment by Feliks151450** [comment]
> 我用的是第三方的API，是支持文件的，上传方法同图片，使用base64，所以想问一下从哪里能修改

gpt-4-all？还没见过直接用base64上传文件的，都是传到一个存储的地方转成链接再让gpt读链接

[a comment by an automated account is not shown]

**Comment by cyx1322** [comment]
> > 我用的是第三方的API，是支持文件的，上传方法同图片，使用base64，所以想问一下从哪里能修改
> 
> gpt-4-all？还没见过直接用base64上传文件的，都是传到一个存储的地方转成链接再让gpt读链接

用的pandoratov1api项目做的，直接支持base64，gpt-4-all我感觉不太稳定

[a comment by an automated account is not shown]

**Comment by arvinxx** [comment, marked as answer]
> 我用的是第三方的API，是支持文件的，上传方法同图片，使用base64，所以想问一下从哪里能修改

试试注释这一行：

https://github.com/lobehub/lobe-chat/blob/main/src/features/ChatInput/ActionBar/FileUpload.tsx#L25

另外我们的展示上没有对非图片类的文件做兼容，所以可能预览会挂掉。你可以测试下看看行不行，最好结果也能反馈一下给我。

[a comment by an automated account is not shown]

**Comment by Feliks151450** [comment]
> > 我用的是第三方的API，是支持文件的，上传方法同图片，使用base64，所以想问一下从哪里能修改
> 
> 试试注释这一行：
> 
> https://github.com/lobehub/lobe-chat/blob/main/src/features/ChatInput/ActionBar/FileUpload.tsx#L25
> 
> 另外我们的展示上没有对非图片类的文件做兼容，所以可能预览会挂掉。你可以测试下看看行不行，最好结果也能反馈一下给我。
刚试了一下，确实可以，用的逆向的gpt-4-all，但是文件不能太大（会报response.413），传了个html文件过去，能识别
<img width="578" alt="image" src="https://github.com/lobehub/lobe-chat/assets/44024139/ffc741c0-24a4-4196-b726-632c3e3dde11">

**Reply by arvinxx** [reply]
还真能用啊…😅 看来我当时做成通用文件的读取和加载还挺正确的。

想问下，有没有试过 zip 包？它能不能支持解压和读取哇

**Reply by sweetcard** [reply]
要是把预览改下就好了。哪怕是换个图片或者点击可以下载到本地也是不错的。👍

**Reply by Feliks151450** [reply]
你别说，gpt-4-all还真有点强，我传了一个压缩包问他里面有什么，他直接用python写了个解压代码运行，然后分析了运行结果
<img width="632" alt="image" src="https://github.com/lobehub/lobe-chat/assets/44024139/41bf4a1d-6a90-48fe-8b39-88f735d157a6">
<img width="606" alt="image" src="https://github.com/lobehub/lobe-chat/assets/44024139/97116d9d-bc4b-40f7-9a3d-89b1946ee166">

**Reply by arvinxx** [reply]
太 6 了

**Comment by cyx1322** [comment]
> > > 我用的是第三方的API，是支持文件的，上传方法同图片，使用base64，所以想问一下从哪里能修改
> > 
> > 
> > 试试注释这一行：
> > https://github.com/lobehub/lobe-chat/blob/main/src/features/ChatInput/ActionBar/FileUpload.tsx#L25
> > 另外我们的展示上没有对非图片类的文件做兼容，所以可能预览会挂掉。你可以测试下看看行不行，最好结果也能反馈一下给我。
> > 刚试了一下，确实可以，用的逆向的gpt-4-all，但是文件不能太大（会报response.413），传了个html文件过去，能识别
> 
> <img alt="image" width="578" src="https://private-user-images.githubusercontent.com/44024139/293926860-ffc741c0-24a4-4196-b726-632c3e3dde11.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQyODk5NjUsIm5iZiI6MTcwNDI4OTY2NSwicGF0aCI6Ii80NDAyNDEzOS8yOTM5MjY4NjAtZmZjNzQxYzAtMjRhNC00MTk2LWI3MjYtNjMyYzNlM2RkZTExLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTAzVDEzNDc0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE2NTgwODQ1Zjc3NmVmZDg1MjgwYzMwN2Q0MjE0ODU4OWUwZjMwY2VhMzQ2M2NkYjhlZGJmMTY2NmNkNGE4MjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.0d_NP1la2lY9pWln_6lgL0KlrOt21t9glZG4Wuse2fI">

你用的哪个API提供商？ godgpt？他们老板说不打算对官方API做太多兼容，所以base64只支持大概1mb

[a comment by an automated account is not shown]

**Comment by cyx1322** [comment]
> > 我用的是第三方的API，是支持文件的，上传方法同图片，使用base64，所以想问一下从哪里能修改
> 
> 试试注释这一行：
> 
> https://github.com/lobehub/lobe-chat/blob/main/src/features/ChatInput/ActionBar/FileUpload.tsx#L25
> 
> 另外我们的展示上没有对非图片类的文件做兼容，所以可能预览会挂掉。你可以测试下看看行不行，最好结果也能反馈一下给我。

我刚才测试了一下，PDF直接调用系统PDF阅读器显示
![IMG_1826](https://github.com/lobehub/lobe-chat/assets/81369393/acc54777-629b-47b9-98e7-a27dd4d91c53)
但其他文件格式就是一个灰色的图片，就像上边的图片一样

**Comment by Feliks151450** [comment]
> > > 我用的是第三方的API，是支持文件的，上传方法同图片，使用base64，所以想问一下从哪里能修改
> > 
> > 
> > 试试注释这一行：
> > https://github.com/lobehub/lobe-chat/blob/main/src/features/ChatInput/ActionBar/FileUpload.tsx#L25
> > 另外我们的展示上没有对非图片类的文件做兼容，所以可能预览会挂掉。你可以测试下看看行不行，最好结果也能反馈一下给我。
> 
> 我刚才测试了一下，PDF直接调用系统PDF阅读器显示 ![IMG_1826](https://private-user-images.githubusercontent.com/81369393/293929561-acc54777-629b-47b9-98e7-a27dd4d91c53.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQyOTA4NDgsIm5iZiI6MTcwNDI5MDU0OCwicGF0aCI6Ii84MTM2OTM5My8yOTM5Mjk1NjEtYWNjNTQ3NzctNjI5Yi00N2I5LTk4ZTctYTI3ZGQ0ZDkxYzUzLmpwZWc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEwM1QxNDAyMjhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02YzNlYTVhMmU1NzExMzliY2IwNDc2ZTliNmRkMDllMTRmYzIwZjFmM2E3YTE1MmUwMzJmYTNhY2E3OTMzZDZiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.g0QBSH9x2sgtmJjgTNaQdqgkUYJOFB3BV7tIdXbD3XI) 但其他文件格式就是一个灰色的图片，就像上边的图片一样

PDF的显示似乎和浏览器有关，我用的arc，上传的pdf只能是一个灰色图片

[a comment by an automated account is not shown]
